### PR TITLE
TYP: freq and na_value

### PR DESCRIPTION
**Pull request by twoertwein**
Type annotations are only related by having `lib.no_default` as a default value (pyright then assumes that their type is `lib.NoDefault`).

### CODE REVIEW

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -264,7 +264,10 @@ def _validate_dtype(cls, values, dtype):
     # error: Signature of "_simple_new" incompatible with supertype "NDArrayBacked"
     @classmethod
     def _simple_new(  # type: ignore[override]
-        cls, values: np.ndarray, freq: BaseOffset | None = None, dtype=DT64NS_DTYPE
+        cls,
+        values: np.ndarray,
+        freq: str | BaseOffset | lib.NoDefault | None = None,
```
Comment (mroeschke) [inline pandas/core/arrays/datetimes.py:269]: @jbrockmendel should the caller ensure the `freq` is not a string when calling `_simple_new`?

Comment (jbrockmendel) [reply pandas/core/arrays/datetimes.py:269]: this should be `BaseOffset | None`

Comment (twoertwein) [reply pandas/core/arrays/datetimes.py]: Thank you, I changed it!

File: pandas/core/arrays/datetimes.py [hunk 2]
```diff
@@ -291,7 +294,7 @@ def _from_sequence_not_strict(
         dtype=None,
         copy: bool = False,
         tz=None,
-        freq=lib.no_default,
+        freq: str | BaseOffset | lib.NoDefault | None = lib.no_default,
```
Comment (mroeschke) [inline pandas/core/arrays/datetimes.py:297]: Looks like `str` reappeared here

Comment (twoertwein) [reply pandas/core/arrays/datetimes.py:297]: Thanks, I will look into that! I had previously changed only `_simple_new` but not `_from_sequence_not_strict`.

Comment (mroeschke) [reply pandas/core/arrays/datetimes.py:297]: Oh sorry! From the diff I thought this was still under `_simple_new`, I think it's okay if freq is a `str` here

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @twoertwein

**Pull request merged**